src/netspresso_trainer/models/op/custom.py

```diff
@@ -489,3 +489,142 @@ def forward(self, x: Tensor) -> Tensor:
 
     # def __repr__(self):
     #     return "{}(type={})".format(self.__class__.__name__, self.pool_type)
+
+
+class Focus(nn.Module):
+    """Focus width and height information into channel space."""
+
+    def __init__(self, in_channels, out_channels, ksize=1, stride=1, act_type="silu"):
+        super().__init__()
+        self.conv = ConvLayer(in_channels=in_channels * 4,
+                              out_channels=out_channels, 
+                              kernel_size=ksize, 
+                              stride=stride, 
+                              act_type=act_type)
+
+    def forward(self, x):
+        # shape of x (b,c,w,h) -> y(b,4c,w/2,h/2)
+        patch_top_left = x[..., ::2, ::2]
+        patch_top_right = x[..., ::2, 1::2]
+        patch_bot_left = x[..., 1::2, ::2]
+        patch_bot_right = x[..., 1::2, 1::2]
+        x = torch.cat(
+            (
+                patch_top_left,
+                patch_bot_left,
+                patch_top_right,
+                patch_bot_right,
+            ),
+            dim=1,
+        )
+        return self.conv(x)
+
+
+class CSPLayer(nn.Module):
+    """C3 in yolov5, CSP Bottleneck with 3 convolutions"""
+
+    def __init__(
+        self,
+        in_channels,
+        out_channels,
+        n=1,
+        shortcut=True,
+        expansion=0.5,
+        #depthwise=False,
+        act_type="silu",
+    ):
+        """
+        Args:
+            in_channels (int): input channels.
+            out_channels (int): output channels.
+            n (int): number of Bottlenecks. Default value: 1.
+        """
+        # ch_in, ch_out, number, shortcut, groups, expansion
+        super().__init__()
+        hidden_channels = int(out_channels * expansion)  # hidden channels
+        self.conv1 = ConvLayer(in_channels=in_channels, 
+                               out_channels=hidden_channels,
+                               kernel_size=1, 
+                               stride=1, act_type=act_type)
+        self.conv2 = ConvLayer(in_channels=in_channels,
+                              out_channels=hidden_channels, 
+                              kernel_size=1, 
+                              stride=1, act_type=act_type)
+        self.conv3 = ConvLayer(in_channels=2 * hidden_channels, 
+                               out_channels=out_channels, 
+                               kernel_size=1, 
+                               stride=1, act_type=act_type)
+        
+        # Newly define because of slight difference with Bottleneck of custom.py
+        class DarknetBlock(nn.Module):
```

Comment:
@illian01 

Should it be internally defined? It seems like no issue but is quite confusing to others...

Reply:
It doesn't need to be inside of `CSPLayer`. 
I thought it is less confusing ... but that may be because I wrote that code.
I will move `DarknetBlock` to outside.

Reply:
@illian01 

Actually, I should not ask about this cause' our ruff check doesn't say anything :)
Sorry for confusion, but it would be better if we get some code style guideline 👍🏻